usr/share/rear/lib/_framework-setup-and-functions.sh

```diff
@@ -1507,3 +1507,324 @@ function ProgressInfo () {
     : ;
 }
 
+# Sourcing functions to enforce trusted sourcing:
+
+# Specify default TRUSTED_OWNERS:
+#
+# The owner of sbin/rear is always trusted.
+# Usually the owner of sbin/rear is 'root'
+# but if not (e.g. in case of Git checkout on the original system)
+# also 'root' is trusted by default:
+sbin_rear_owner_name="$( stat -L -c %U "$SCRIPT_FILE" )"
+test "$sbin_rear_owner_name" = "root" && TRUSTED_OWNERS=( 'root' ) || TRUSTED_OWNERS=( 'root' "$sbin_rear_owner_name" )
+# 'readonly TRUSTED_OWNERS' happens in sbin/rear
+# after the normal user configuration files (site.conf local.conf rescue.conf) were sourced
+# so that the user can specify what he needs, e.g. for whatever third-party (backup) software.
+
+# Check file owner is trusted
+# see https://relax-and-recover.org/documentation/security-architecture
+function is_trusted_owner () {
+    local file="$1"
+    local owner_name=""
+    local trusted_owner=""
+    # Do not error out in 'stat' when it is neither a regular file nor a link to a regular file
+    # but it is not trusted when it is neither a regular file nor a link to a regular file:
+    test -f "$file" || return 1
+    # Empty TRUSTED_OWNERS means all regular file owners are blindly trusted,
+    # cf https://github.com/rear/rear/pull/3379#issuecomment-2633123866
+    test ${#TRUSTED_OWNERS[@]} -eq 0 && return 0
```

Comment:
```suggestion
    # Empty TRUSTED_OWNERS means all regular file owners are blindly trusted,
    # cf https://github.com/rear/rear/pull/3379#issuecomment-2633123866
    test ${#TRUSTED_OWNERS[@]} -eq 0 && return 0
    # Do not error out in 'stat' when it is neither a regular file nor a link to a regular file
    # but it is not trusted when it is neither a regular file nor a link to a regular file:
    test -f "$file" || return 1
```

test first for common case

Reply:
Thank you!
This was clearly the wrong ordering.